.pre_commit/generate-facets/generate.py

```diff
@@ -0,0 +1,256 @@
+    """Deep merges two dictionaries.
+
+    This function merges two dictionaries while handling nested dictionaries.
+    For keys that exist in both dictionaries, the values from dict2 take precedence.
+    If a key exists in both dictionaries and the values are dictionaries themselves,
+    they are merged recursively.
+    """
+    merged = dict1.copy()
+    for k, v in dict2.items():
+        if k in merged and isinstance(v, dict):
+            merged[k] = deep_merge_dicts(merged.get(k, {}), v)
+        else:
+            merged[k] = v
+    return merged
+
+
+def load_specs(base_spec_location: pathlib.Path, facets_spec_location: pathlib.Path) -> list[pathlib.Path]:
+    """Load base `OpenLineage.json` and other facets' spec files"""
+    locations = []
+
+    file_specs = [
+        base_spec_location.resolve(),  # Base spec must be the first element
+        *sorted(pathlib.Path(os.path.abspath(facets_spec_location)).glob("*.json")),
+    ]
+    for file_spec in file_specs:
+        spec = json.loads(file_spec.read_text())
+        parse_additional_data(spec, file_spec.name)
+        locations.append(file_spec)
+
+    return locations
+
+
+def parse_additional_data(spec, file_name):
+    """Parse additional data from spec files.
+
+    Parses:
+        * schema URLs
+        * base IDs
+    """
+    base_id = spec["$id"]
+    for name, _ in spec["$defs"].items():
+        SCHEMA_URLS[name] = f"{base_id}#/$defs/{name}"
+        BASE_IDS[file_name] = spec["$id"]
+
+
+def parse_and_generate(locations):
+    """Parse and generate data models from a given specification."""
+
+    extra_schema_urls = {obj_name: {"_schemaURL": schema_url} for obj_name, schema_url in SCHEMA_URLS.items()}
+    extra_redact_fields = {obj_name: {"redact_fields": fields} for obj_name, fields in REDACT_FIELDS.items()}
+
+    temporary_locations = []
+    with tempfile.TemporaryDirectory() as tmp:
+        tmp_directory = pathlib.Path(tmp).resolve()
+        for location in locations:
+            tmp_location = (tmp_directory / location.name).resolve()
+            tmp_location.write_text(location.read_text())
+            temporary_locations.append(tmp_location)
+
+        os.chdir(tmp_directory)
+        # first parse OpenLineage.json
+        parser = JsonSchemaParser(
+            source=temporary_locations[:1],
+            data_model_type=NEW_MODEL,
+            data_model_root_type=DATA_MODELS_TYPES.root_model,
+            data_model_field_type=DATA_MODELS_TYPES.field_model,
+            data_type_manager_type=DATA_MODELS_TYPES.data_type_manager,
+            dump_resolve_reference_action=DATA_MODELS_TYPES.dump_resolve_reference_action,
+            special_field_name_prefix="",
+            use_schema_description=True,
+            field_constraints=True,
+            use_union_operator=True,
+            use_standard_collections=True,
+            base_class="openlineage.client.utils.RedactMixin",
+            class_name="ClassToBeSkipped",
```

Comment:
Yes, in `dataclass.jinja2` I'm using this name to skip adding root model. We don't need it after all.
The example for such model is:
```python
@define
class Model(RedactMixin):
    sql: SQLJobFacet | None = field(default=None)
```

so you don't manually put `SQLJobFacet` under `sql` key. This would more refactoring for sure so for now I'm skipping it.